src/common.c

```diff
@@ -2788,152 +3003,65 @@ static int stlink_write_option_bytes_gx(stlink_t *sl, uint8_t* base, stm32_addr_
     stlink_write_debug32(sl, STM32Gx_FLASH_CR, val);
 
-    /* Re-lock option bytes */
-    stlink_read_debug32(sl, STM32Gx_FLASH_CR, &val);
-    val |= (1u << STM32Gx_FLASH_CR_OPTLOCK);
-    stlink_write_debug32(sl, STM32Gx_FLASH_CR, val);
-
     /* Re-lock flash. */
+    lock_flash_option(sl);
     lock_flash(sl);
 
     return 0;
 }
 
-
-/**
- * Write option bytes
- * @param sl
- * @param addr of the memory mapped option bytes
- * @param base option bytes to write
- * @return 0 on success, -ve on failure.
- */
-static int stlink_write_option_bytes_l0_cat2(stlink_t *sl, uint8_t* base, stm32_addr_t addr, uint32_t len) {
-
-    uint32_t val;
-    (void) addr;
-    (void) len;
-
-    stlink_read_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, &val);
-    if (val & STM32L0_FLASH_PELOCK_BIT) {
-        WLOG("Unlocking flash\n");
-        //Unlock data EEPROM and the FLASH_PECR register (reference page 74)
-        stlink_write_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PEKEYR_OFF, 0x89ABCDEF);
-        stlink_write_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PEKEYR_OFF, 0x02030405);
-
-        stlink_read_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, &val);
-        if (val & STM32L0_FLASH_PELOCK_BIT) {
-            ELOG("Flash unlock failed! System reset required to be able to unlock it again!\n");
-            return -1;
-        }
-    }
-
-    stlink_read_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, &val);
-    if ((val & (STM32L0_FLASH_OPTLOCK_BIT))) {
-        WLOG("Unlocking options\n");
-        //Unlock the Option bytes area (reference page 76)
-        stlink_write_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_OPTKEYR_OFF, 0xFBEAD9C8);
-        stlink_write_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_OPTKEYR_OFF, 0x24252627);
-
-        stlink_read_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, &val);
-        if (val & STM32L0_FLASH_OPTLOCK_BIT) {
-            ELOG("Options unlock failed! System reset required to be able to unlock it again!\n");
-            return -1;
-        }
-    }
-
-    /* Write options bytes */
-    uint32_t data;
-    write_uint32((unsigned char*) &data, *(uint32_t*) (base));
-    WLOG("Writing option bytes 0x%04x\n", data);
-    stlink_write_debug32(sl, STM32_L0_CATx_OPTION_BYTES_BASE, data);
-
-    /* Reload options */
-    stlink_read_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, &val);
-    val |= (STM32L0_FLASH_OBL_LAUNCH_BIT);
-    stlink_write_debug32(sl, STM32L0_FLASH_REGS_ADDR + FLASH_PECR_OFF, val);
-
-    return 0;
-}
-
 /**
  * Write option bytes
  * @param sl
  * @param addr of the memory mapped option bytes
  * @param base option bytes to write
  * @return 0 on success, -ve on failure.
  */
-static int stlink_write_option_bytes_l1(stlink_t *sl, uint8_t* base, stm32_addr_t addr, uint32_t len) {
-
+static int stlink_write_option_bytes_l0(stlink_t *sl, uint8_t* base, stm32_addr_t addr, uint32_t len)
+{
+	uint32_t flash_base = get_stm32l0_flash_base(sl);
```

Comment:
the indentation, if possible, should be the same as the next line.

Reply:
Please use 4 spaces for indentation instead of tab spacing. (I know this is already inconsistent, but will be aligned throughout the codebase midterm.) Also there will be some code style guidelines soon...

Reply:
Oops

Reply:
No worries, it would just be convenient to save some upcoming work where possible.